src/fs.rs

```diff
@@ -963,6 +967,25 @@ mod tests {
         assert_eq!(m, mime::APPLICATION_OCTET_STREAM);
     }
 
+    #[test]
+    fn test_if_modified_since_ignored() {
+        let mut file = NamedFile::open("Cargo.toml")
+            .unwrap()
+            .set_cpu_pool(CpuPool::new(1));
+        let since = header::HttpDate::from(
```

Comment:
Sorry, I forgot one thing: could you please check if we have positive test case for `If-Modified-Since` if not, I'd appreicate if you could add it.

i.e. the same test but with no `IF_NONE_MATCH` header

Reply:
No problem, I added a positive test case for the `If-Modified-Since`, pls check

Reply:
Thanks, but it would be nice if the `If-None-Match` test would verify that we're actually ignoring `If-Modified-Since`
Now that you have both tests you should see that both results in the same outcome, could you please adjust `since` value so that it could result in sending back file and having `If-None-Match` to prevent that? 😄